lib/Dialect/Torch/Transforms/ScalarizeShapes.cpp

```diff
@@ -1065,6 +1215,34 @@ bool isAnchorOp(Operation *op) {
          isPrimListOfInts(op);
 }
 
+// The argument to this function, op, is the use of some source op, srcOp. If
+// this function returns true, we want to invalidate srcOp as a target for shape
+// scalarization.
+bool isInvalidValidViewConsumer(Operation *op,
+                                SetVector<Operation *> &workList) {
+  // if the consumer isn't a view op, don't invalidate it
+  auto view = dyn_cast_or_null<AtenViewOp>(op);
+  if (!view)
+    return false;
+  auto resultTy = dyn_cast<ValueTensorType>(view.getType());
+  if (!resultTy || !resultTy.hasDtype())
+    return true;
+  // if the view op doesn't return integer types, then srcOp is not a shape
+  // tensor. note: prim lists will always get added before reaching this
+  // function call.
+  if (!isa<mlir::IntegerType>(resultTy.getDtype()))
+    return true;
+  // check uses of the view op.
+  // If the view op has a use in our worklist, then it needs to be scalarized.
+  for (OpOperand &use : op->getUses()) {
+    Operation *userOp = use.getOwner();
+    if (workList.contains(userOp))
+      return false;
+  }
```

Comment:
I'm a bit confused why this is needed, shouldn't the `root` `AtenViewOp` be omitted from the worklist instead?

Reply:
I need to add view ops in general since I want to apply the canonicalization pattern to all view ops in the IR. In the near future, I think we should move the canonicalization pattern out of this pass, since handling these somewhat unrelated view ops is annoying.